kratos/geometries/geometry_data.h

```diff
@@ -153,6 +152,29 @@ class GeometryData
         Kratos_Quadrature_Point_Surface_In_Volume_Geometry,
         NumberOfGeometryTypes // Note that this entry needs to be always the last to be used as geometry types counter
     };
+
+    /**
+     * @enum KratosGeometryOrderType
+     * @brief Defines the order of geometries in Kratos.
+     * @details This enumeration lists the different polynomial orders that can be used for defining the geometry of elements in Kratos. The geometry order impacts the accuracy and computational cost of simulations.
+     * @var KratosGeometryOrderType::Kratos_Zero_Order: Defines a zero-order geometry. This is 0D geometry, aka discrete points, and is the simplest form of geometry.
+     * @var KratosGeometryOrderType::Kratos_Linear_Order: Defines a linear geometry. This is the simplest form, with minimal computational cost but lower accuracy.
+     * @var KratosGeometryOrderType::Kratos_Quadratic_Order: Defines a quadratic geometry. Offers a balance between computational cost and accuracy.
+     * @var KratosGeometryOrderType::Kratos_Cubic_Order: Defines a cubic geometry. Higher computational cost than quadratic, but offers improved accuracy.
+     * @var KratosGeometryOrderType::Kratos_Quartic_Order: Defines a quartic geometry. Further increases computational cost and accuracy.
+     * @var KratosGeometryOrderType::Kratos_Quintic_Order: Defines a quintic geometry. Represents the highest polynomial order here, with the highest accuracy and computational cost.
+     * @var KratosGeometryOrderType::Kratos_Unknown_Order: Used when the geometry order is unknown or not specified. Allows for flexibility in geometry definition.
+    */
+    enum KratosGeometryOrderType {
```

Comment:
this will be **very** difficult to use. I'd rather recommend an optional<int>

Reply:
but why go the roundabout way of defining it as an `enum` in the first place? What does an `enum` do here that an `std::optional<int>` cannot? As far as I see it, the `enum` arbitrarily restricts the orders we can represent.

Reply:
We already use enums in many things for geometries, would you thinks it would be easier to just simply use unsigned int and end of story? opinions @KratosMultiphysics/technical-committee

Reply:
> would you thinks it would be easier to just simply use unsigned int and end of story

an `unisgned` wouldn't be able to represent undefined order, and I think the undefined state of an `std::optional` fits better than using `-1` with signed integers, or some other arbitrary stuff.

Reply:
Okay

Reply:
@jcotela do you agree?, opinion  @KratosMultiphysics/technical-committee ?

Reply:
Hi @loumalouomega sorry for the delay, I had forgotten about this PR. Unfortunately, this change was intended for a feature that we won't be implementing so strictly speaking I no longer need it. Having said that, I think it may be useful to have a way to programmatically inquire about the interpolation order of a specific geometry instance, if the core developers agree.

Regarding the enum vs std::optional, I understand that the order is a category here (the geometry is lineal or quadratic in the same way it is a linear triangle or a cubic hexhahedron) so an enum seems appropriate. I could see using an int so that values have a natural greater/less than comparison in addition to equality checks, but I am not sure about the need for optional. After all, it should be straightforward to figure out the interpolation order for each derived geometry. @matekelemen what use case do you see for an undefined order? Working with base geometery instances?

Reply:
I will not touch this PR until a consencus is reached

Reply:
> what use case do you see for an undefined order?

Well, there's a `Kratos_Unknown_Order` in the current PR so I thought there's a use case for it. Also geometries that don't overload this function should return an undefined value instead of some made-up default.

I'm not familiar with IGA in kratos, but I could imagine an undefined order for a default-constructed IGA geometry as well.